contracts/VAIController.sol

```diff
@@ -402,6 +416,17 @@ contract VAIController is VAIControllerStorageG2, VAIControllerErrorReporter, Ex
                 return (uint(Error.MATH_ERROR), 0);
             }
 
+            (, uint collateralFactorMantissa,) = Comptroller(address(comptroller)).markets(address(enteredMarkets[i]));
+            (vars.mErr, vars.sumSupply) = mulUInt(vars.sumSupply, collateralFactorMantissa);
```

Comment:
Wait, I think you should only include `enteredMarkets[i]` only here, but now you do sth like

Market2.collateralFactor * (Market1.collateralFactor * supply1 + supply2))

but it should be

Market1.collateralFactor * supply1 + Market2.collateralFactor * supply2

Reply:
@kkirka Fixed it. Now I am taking weighted supply of each market and summing them up to find the total weighted supply.